### PR TITLE
grass: Compile with libLAS

### PR DESCRIPTION
**Pull request by mpickering**
###### Motivation for this change

This enables the grass commands `r.in.lidar` and so on for working with lidar data. 

Thanks to @LnL7 for all his help with these changes. 

Subsumes #29082

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by joachifm** [comment]
@GrahamcOfBorg build grass

File: pkgs/development/libraries/libLAS/default.nix [hunk 1]
```diff
@@ -9,14 +9,22 @@ stdenv.mkDerivation rec {
...
 
-  buildInputs = [ boost cmake gdal libgeotiff libtiff LASzip ];
+  buildInputs = [ boost cmake gdal libgeotiff libtiff LASzip ]
+                ++ stdenv.lib.optional stdenv.isDarwin fixDarwinDylibNames;
 
+  cmakeFlags = [
+    "-DGDAL_CONFIG=${gdal}/bin/gdal-config"
+  ];
+
+  postFixup = ''
```
Comment (LnL7) [inline pkgs/development/libraries/libLAS/default.nix]: `stdenv.lib.optionalString stdenv.isDarwin`

**Review by LnL7 — COMMENTED** [review]
Looks good otherwise

File: pkgs/development/libraries/libLAS/default.nix (hunk 1 above)
Comment (mpickering) [reply pkgs/development/libraries/libLAS/default.nix]: Fixed.

**Comment by LnL7** [comment]
@GrahamcOfBorg build grass

**Review by LnL7 — APPROVED** [review]

**Pull request merged**